cmake 2.8.10:  Cannot compile with the user supplied flags

cmake 2.8.10 seems to be unable to configure starting around #2822 : 

```
CMake Error at cmake/setup_compiler_flags.cmake:83 (MESSAGE):
  

    Configuration error: Cannot compile with the user supplied flags:
      CXX flags (DEBUG):  
      LD flags  (DEBUG):  
      LIBRARIES (DEBUG): ;
    Please check the CMake variables
      DEAL_II_CXX_FLAGS, DEAL_II_CXX_FLAGS_DEBUG,
      DEAL_II_LINKER_FLAGS, DEAL_II_CXX_FLAGS_DEBUG
    and the environment variables CXXFLAGS, LDFLAGS.

  

Call Stack (most recent call first):
  cmake/macros/macro_verbose_include.cmake:19 (INCLUDE)
  CMakeLists.txt:95 (VERBOSE_INCLUDE)
```

--- a/cmake/macros/macro_check_compiler_setup.cmake
+++ b/cmake/macros/macro_check_compiler_setup.cmake
@@ -25,7 +25,13 @@
 # ${_var} is set to true, otherwise it is set to false.
 #
 
-MACRO(CHECK_COMPILER_SETUP _compiler_flags _linker_flags _var)
+MACRO(CHECK_COMPILER_SETUP _compiler_flags_unstr _linker_flags_unstr _var)
+  #
+  # Strip leading and trailing whitespace to make CMake 2.8.8 happy
+  #
+  STRING(STRIP "${_compiler_flags_unstr}" _compiler_flags)
+  STRING(STRIP "${_linker_flags_unstr}" _linker_flags)
+
   #
   # Rerun this test if flags have changed:
   #

